src/utils/userLocation.ts

```diff
@@ -1,52 +1,83 @@
-import firebase, { UserInfo } from "firebase/app";
+import firebase from "firebase/app";
 
 import { AnyRoom } from "types/rooms";
 
 import { updateUserProfile } from "pages/Account/helpers";
 import { useInterval } from "hooks/useInterval";
 
 import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
+import { openRoomUrl } from "./url";
 
 const LOCATION_INCREMENT_SECONDS = 10;
 const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
 
 type LocationData = Record<string, number>;
 
-export const updateLocationData = (
-  userId: string,
-  newLocationData: LocationData = {}
-) => {
+type UpdateLocationDataProps = {
+  userId: string;
+  newLocationData: LocationData;
+};
+
+export const updateLocationData = ({
+  userId,
+  newLocationData,
+}: UpdateLocationDataProps) => {
   updateUserProfile(userId, {
     lastSeenAt: getCurrentTimeInMilliseconds(),
     lastSeenIn: newLocationData,
   });
 };
 
-export interface TrackLocationProps {
+type SetNewLocationDataProps = {
   userId: string;
   locationName: string;
   lastSeenIn?: LocationData;
-}
+};
 
-export const trackLocationEntered = ({
+export const setNewLocationData = ({
```

Comment:
The naming of this feels weird.. almost reversed. I would expect that `setNewLocationData` would override everything, and `updateLocationData` would probably do the more selective update.

Would probably rename `setNewLocationData` to just `setLocationData` to match the same 'format' of `updateLocationData` as well.

(Doesn't necessarily have to follow the above, but the distinction/usages of them feels a bit confusing at the moment, so if we can improve that, that would be my aim here)